### ISSUE TITLE
Include Dashdb (DB2) drivers in the node.js (and other) runtimes

### ISSUE THREAD

**Issue by mpuittinen** [apache/incubator-openwhisk#1932]
We're developing a customer solution on Bluemix that need to access a Dashdb database. Our current workaround is to create a docker image with the drivers (not sure whether license allows that), but the cold start time is absolutely horrible (20secs).

Please include the DB2 driver binaries (and Node modules) in the standard Node.js Openwhisk runtime.

**Comment by jthomas** [comment]
@mpuittinen Have you looked at bundling up the drivers in a zipped package? 
http://jamesthom.as/blog/2016/11/28/npm-modules-in-openwhisk/

**Comment by glikson** [comment]
@jthomas this particular set of libraries seem to be too large for a zipped action (in the order of 100MB).
@mpuittinen it is probably not feasible to include all the possible client libraries in a single base image.. maybe if you base your image on a standard OpenWhisk image, pulling it from docker hub should attempt reusing the previous layers - making the cold start potentially shorter (and potentially happen only once for any action using this particular set of client libraries).

**Comment by rabbah** [comment]
With the latest simplifications of action "kinds", I expect we can add a new kind eg `nodejs:6-dashb` as a deployment configuration. This will intersect with work that @csantanapr has started to publish runtime images to dockerhub. I imagine that we can setup individual repos for various runtimes, push them to dockerhub, and point the [runtime manifest](https://github.com/openwhisk/openwhisk/pull/1980/files#diff-2d879ae862e88c1fe4c4e19affaeffddR38) at these to extend the runtime support in a particular deployment.

**Comment by mpuittinen** [comment]
What kind of caching mechanism is there for the images from dockerhub? i.e. does that introduce a delay for cold starts with large images? I’ve entered another ticket to have support for Bluemix Container Service registries, would that improve cold start times?

**Comment by rabbah** [comment]
@csantanapr FYI.
Closing as won't fix. Base packages are specific to a deployment.
